sumologic/resource_sumologic_polling_source.go

```diff
@@ -75,19 +75,69 @@ func resourceSumologicPollingSource() *schema.Resource {
 		MaxItems: 1,
 		Elem: &schema.Resource{
 			Schema: map[string]*schema.Schema{
+				"type": {
+					Type:         schema.TypeString,
+					Required:     true,
+					ValidateFunc: validation.StringInSlice([]string{"S3BucketPathExpression", "CloudWatchPath"}, false),
+				},
 				"bucket_name": {
 					Type:     schema.TypeString,
-					Required: true,
```

Comment:
Are there some kind of validation we would lose in Terraform after making `bucket_name` and `path_expression` optional because now we have two path types?

Reply:
not that I know of